### PR TITLE
fix(chat): fix clip icon appearance (Issue #2063)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

If the user application has the "Max. attachments count" field equal to 0, then the clip icon in the chat should not be displayed, it is displayed if it is a number or empty (if empty it takes the maximum integer value)

Issues:

- Issue #2063 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2093.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Common/ApplicationDialog.tsx [hunk 1]
```diff
@@ -566,9 +570,8 @@ const ApplicationDialogView: React.FC<Props> = ({
               className={inputClassName}
               placeholder={t('Enter the maximum number of attachments') || ''}
               onChange={(e) => {
-                const value = e.target.value
-                  ? Number(e.target.value)
-                  : undefined;
+                const value =
+                  e.target.value !== '' ? Number(e.target.value) : undefined;
                 if (!e.target.value || Number.isSafeInteger(value)) {
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationDialog.tsx]:
```suggestion
                if (!value || Number.isSafeInteger(value)) {
```

File: apps/chat/src/components/Common/ApplicationDialog.tsx [hunk 2]
```diff
@@ -118,7 +118,7 @@ const ApplicationDialogView: React.FC<Props> = ({
   const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
   const [isPublishing, setIsPublishing] = useState(false);
   const [maxInputAttachmentsValue, setMaxInputAttachmentsValue] = useState(
-    selectedApplication?.maxInputAttachments || undefined,
+    selectedApplication?.maxInputAttachments ?? undefined,
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationDialog.tsx]:
```suggestion
    selectedApplication?.maxInputAttachments,
```

File: apps/chat/src/components/Common/ApplicationDialog.tsx [hunk 3]
```diff
@@ -568,9 +572,8 @@ const ApplicationDialogView: React.FC<Props> = ({
               className={inputClassName}
               placeholder={t('Enter the maximum number of attachments') || ''}
               onChange={(e) => {
-                const value = e.target.value
-                  ? Number(e.target.value)
-                  : undefined;
+                const value =
+                  e.target.value !== '' ? Number(e.target.value) : undefined;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationDialog.tsx]: Are you sure that we handle spaces and NaN cases?
image

Comment (Derikyan) [reply apps/chat/src/components/Common/ApplicationDialog.tsx]: onChange={(e) => {
    const numericValue = e.target.value.replace(/[^0-9]/g, '');
    const value = numericValue !== '' ? Number(numericValue) : undefined;
                if (!value || Number.isSafeInteger(value)) {
                  setMaxInputAttachmentsValue(value);
                }
                handleChangeHandlerAttachments?.(e);
              }}
              
I changed the code, now it doesn't let enter space or symbols other than numbers

**Review by IlyaBondar — APPROVED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/515598)

**Pull request merged**